src/azure-cli/azure/cli/command_modules/network/_params.py

```diff
@@ -1348,8 +1348,27 @@ def load_arguments(self, _):
         c.argument('filters', type=get_json_object)
 
     with self.argument_context('network watcher flow-log') as c:
+        c.argument('network_watcher_name', options_list='--watcher')
+        c.ignore('location')
+        c.argument('flow_log_name', name_arg_type, help='The name of the flow logger', min_api='2019-11-01')
         c.argument('nsg', help='Name or ID of the network security group.')
-        c.argument('enabled', arg_type=get_three_state_flag())
+        c.argument('enabled', arg_type=get_three_state_flag(), help='Enable logging', default='true')
+        c.argument('retention', type=int, help='Number of days to retain logs')
+        c.argument('storage_account', help='Name or ID of the storage account in which to save the flow logs')
+        c.argument('storage_account_id', help='ID of the storage account in which to save the flow logs')
```

Comment:
Why need `storage_accoun`t and `storage_account_id` ?
- `storage_account_id` support name and ID. when given name, CLI will leverage the the value of `--resource-group` to assemble a storage account ID.
- `storage_account_id` supports ID only.

When creating _**network watcher flow-log create**_, the parameter `--resource-group` is used to tell which resource group the `storage_account` and `workspace` they belongs to, instead of the Network Watcher. The Network Watcher's resource group will always be *NetworkWatcherRG*, and the Watcher is identified by location only.

When updating, _**network watcher flow-log**_, the `--resource-group` is used to tell which location of the watcher and the flow-log is. It doesn't have the functionality to restrict the Storage Account and Workspace.

So, another parameter to represent ID only is necessary I think.